### ISSUE TITLE
Error: unknown value_type ()

### ISSUE THREAD

**Issue by sfranzyshen** [wasm3/wasm3#251]
Arduino ESP8266 environment:

I am trying to use wasm3 from within the Arduino environment and I can't seem to get things working ... I am able to build the wasm file (using wasicc ) and I can successfully bootstrap things in the arduino sketch (m3_NewRuntime, m3_NewEnvironment, m3_ParseModule, m3_LoadModule, m3_LinkArduino. m3_FindFunction, and m3_CallV) and I can execute a function ... but when I try and access a function that uses any TYPE other than uint32_t, int32_t ... I get the error in the title above ... are we limited to using only 32 bit types ??

**Comment by sfranzyshen** [comment]
from the files below ... that seems to be what others are doing too ... ???

## wasm3-arduino/main/wasm_apps/cpp/arduino_api.h ##
`WASM_IMPORT("arduino", "digitalWrite")    void digitalWrite   (uint32_t pin, uint32_t value);`

## wasm3-arduino/blob/main/examples/Wasm_Blink/Wasm_Blink.ino ##
```
m3ApiRawFunction(m3_arduino_digitalWrite)
{
    m3ApiGetArg     (uint32_t, pin)
    m3ApiGetArg     (uint32_t, value)

    digitalWrite(pin, value);

    m3ApiSuccess();
}

m3_LinkRawFunction (module, arduino, "digitalWrite",     "v(ii)",  &m3_arduino_digitalWrite);

```
## Arduino.h ##
`void digitalWrite(uint8_t pin, uint8_t val);`

**Comment by vshymanskyy** [comment]
There's not enough info to understand what you're doing here. For example, why `wasicc` is used. Please check out Arduino example, all instructions are included.

**Comment by vshymanskyy** [comment]
@sfranzyshen please provide full steps to reproduce.

**Comment by sfranzyshen** [comment]
I ask a simple question  "are we limited to using only 32 bit types ??" in our code ....

from what I can tell ... the answer is no ... even if it get's changed by the compiler ... it works when using uint8_t and other types in your code ...

sorry for troubling you ...

**Comment by vshymanskyy** [comment]
Yes you can have all other types in Wasm. Many of them will have to be stored in the linear memory, and are passed as an offset (much like a pointer).
